### ISSUE TITLE
No support for RecursiveDo extension?

### ISSUE THREAD

**Issue by Heimdell** [ndmitchell/hlint#58]
I have installed:
hlint-1.8.61
Configuring haskell-src-exts-1.15.0.1

What I do:
```
{-# LANGUAGE RecursiveDo #-}

test = do 
    rec x <- getLine

    return x
```
What I supposed to get:
- ghc compiles;
- hlint - no complain.

What do I get:
- ghc compiles;
- hlint - not:
```
Warning: Parse error in pattern: rec
Found:
    test = do 
  >     rec x <- getLine
    
        return x
```

I use hlint as a part of SublimeHaskell and I have no clue how to pass `-XArrows` to it.

**Comment by jplatte** [comment]
I'm getting the same error, different context. GHC doesn't complain. I found rec used as a normal identifier for a function, and I still get the error message when just typing `rec` into a new `.hs` file and waiting half a second for my editor plugin to execute hlint. The issue on the `haskell-src-exts` project is closed, so I'm unsure if I should have commented there instead.

**Comment by ndmitchell** [comment]
@jplatte The problem is that HLint as standard turns on as many extensions as it can that don't steal "too much" syntax. In your case, the `DoRec` extension steals the `rec` identifier as a keyword. I would generally discourage using `rec` as an identifier because it is a keyword under some extensions, and it is also used to indicate recursive let bindings in the Core language, so it has great scope for confusion. If you do want to continue to use `rec` as a keyword, passing `-XNoDoRec` will disable the conflicting extension, and then your program will parse.

**Comment by ndmitchell** [comment]
As @jplatte mentioned, the ticket was resolved upstream, and I've just tested the original report, and it works fine now, so closing this ticket.

**Comment by jplatte** [comment]
Alright, thanks for the info. I didn't write the code with the rec identifier in it, but it's a project I've started working on recently (haskell-gi). I'll ask the original author about that identifier, but I think it shouldn't be a problem to change it.
